### PR TITLE
Delete unused files

### LINKED ISSUE #2825: [cleanup] Some unused files can be deleted
* be/test/runtime/string_buffer_test.cpp
* be/src/runtime/string_buffer.hpp
* be/src/util/types.h

### PR DESCRIPTION
**Pull request by Ccuurryy**
```
be/src/runtime/string_buffer.hpp
be/src/util/types.h
be/test/runtime/string_buffer_test.cpp
```
The three files are not used and remove them.

### CODE REVIEW

**Comment by Ccuurryy** [comment]
#2825

**Comment by imay** [comment]
@Ccuurryy 
Firstly thank you for you contribution.
You should also delete the corresponding `./runtime/string_buffer_test.cpp` line in `be/test/CMakeLists.txt` file.

**Comment by Ccuurryy** [comment]
@imay 
Sorry to forget that,I push it again

**Comment by imay** [comment]
@Ccuurryy You'd better not to remove file `util/types.h`, because some other files also use the class.

**Comment by stdpain** [comment]
@Ccuurryy 
Since util/types.h is no longer needed, these dependencies need to be removed as well
```
be/src/exprs/slot_ref.cpp:28:#include "util/types.h"
be/src/exprs/anyval_util.h:29:#include "util/types.h"
be/src/exprs/vectorized/function_helper.h:8:#include "util/types.h"
be/src/column/fixed_length_column_base.cpp:14:#include "util/types.h"
be/src/runtime/vectorized/statistic_result_writer.cpp:12:#include "util/types.h"
be/src/runtime/mysql_table_writer.cpp:43:#include "util/types.h"
be/src/runtime/export_sink.cpp:37:#include "util/types.h"
be/src/runtime/memory_scratch_sink.cpp:40:#include "util/types.h"
be/src/runtime/raw_value.h:30:#include "util/types.h"
be/src/runtime/raw_value_ir.cpp:24:#include "util/types.h"
be/src/runtime/file_result_writer.cpp:35:#include "util/types.h"
be/src/runtime/mysql_result_writer.cpp:35:#include "util/types.h"
be/src/runtime/raw_value.cpp:29:#include "util/types.h"
be/src/storage/types.h:51:#include "util/types.h"
be/src/util/types.h:3://   https://github.com/apache/incubator-doris/blob/master/be/src/util/types.h
be/src/util/arrow/row_batch.cpp:47:#include "util/types.h"
be/src/exec/plain_text_builder.cpp:14:#include "util/types.h"
```

**Comment by stdpain** [comment]
@Ccuurryy  you should rebase main branch

**Comment by stdpain** [comment]
some file need change
be/src/runtime/raw_value.h:201

```
+#include "util/unaligned_access.h"
```

```
-    case TYPE_DECIMALV2:
-        return reinterpret_cast<const PackedInt128*>(v1)->value == reinterpret_cast<const PackedInt128*>(v2)->value;
+    case TYPE_DECIMALV2: {
+        int128_t r = unaligned_load<int128_t>(v2);
+        int128_t l = unaligned_load<int128_t>(v1);
+        return r == l;
+    }
```

**Comment by stdpain** [comment]
There are some other areas that also need similar modifications
```
be/src/runtime/raw_value.h:152:        return reinterpret_cast<const PackedInt128*>(v1)->value < reinterpret_cast<const PackedInt128*>(v2)->value;
be/src/runtime/raw_value.h:155:        return reinterpret_cast<const PackedInt128*>(v1)->value < reinterpret_cast<const PackedInt128*>(v2)->value;
be/src/runtime/raw_value.h:209:        return reinterpret_cast<const PackedInt128*>(v1)->value == reinterpret_cast<const PackedInt128*>(v2)->value;
```
Then you should format your code. Try run bash `build-support/clang-format.sh`

**Comment by stdpain** [comment]
@Ccuurryy 
some compile error:
you should use 
```
    case TYPE_DECIMALV2: {
        int128_t r = unaligned_load<int128_t>(v2);
        int128_t l = unaligned_load<int128_t>(v1);
        return r == l;
    }
```
instead of 
```
    case TYPE_DECIMALV2: 
        int128_t r = unaligned_load<int128_t>(v2);
        int128_t l = unaligned_load<int128_t>(v1);
        return r == l;
```

File: be/src/runtime/raw_value.h [hunk 1]
```diff
@@ -147,11 +147,17 @@ inline bool RawValue::lt(const void* v1, const void* v2, const TypeDescriptor& t
     case TYPE_DECIMAL:
         return *reinterpret_cast<const DecimalValue*>(v1) < *reinterpret_cast<const DecimalValue*>(v2);
 
-    case TYPE_DECIMALV2:
-        return reinterpret_cast<const PackedInt128*>(v1)->value < reinterpret_cast<const PackedInt128*>(v2)->value;
+    case TYPE_DECIMALV2: {
+        int128_t r = unaligned_load<int128_t>(v2);
+        int128_t l = unaligned_load<int128_t>(v1);
+        return r == l;
```
Comment (stdpain) [inline be/src/runtime/raw_value.h:153]:
```suggestion
        return r < l;
```

File: be/src/runtime/raw_value.h [hunk 2]
```diff
@@ -198,11 +204,17 @@ inline bool RawValue::eq(const void* v1, const void* v2, const TypeDescriptor& t
     case TYPE_DECIMAL:
         return *reinterpret_cast<const DecimalValue*>(v1) == *reinterpret_cast<const DecimalValue*>(v2);
 
-    case TYPE_DECIMALV2:
-        return reinterpret_cast<const PackedInt128*>(v1)->value == reinterpret_cast<const PackedInt128*>(v2)->value;
+    case TYPE_DECIMALV2: {
+        int128_t r = unaligned_load<int128_t>(v2);
+        int128_t l = unaligned_load<int128_t>(v1);
+        return r < l;
```
Comment (stdpain) [inline be/src/runtime/raw_value.h:210]:
```suggestion
        return r == l;
```

File: be/src/runtime/raw_value.h [hunk 3]
```diff
@@ -147,11 +147,17 @@ inline bool RawValue::lt(const void* v1, const void* v2, const TypeDescriptor& t
...
+        int128_t l = unaligned_load<int128_t>(v1);
+        return r == l;
+    }
 
-    case TYPE_LARGEINT:
-        return reinterpret_cast<const PackedInt128*>(v1)->value < reinterpret_cast<const PackedInt128*>(v2)->value;
+    case TYPE_LARGEINT: {
+        int128_t r = unaligned_load<int128_t>(v2);
+        int128_t l = unaligned_load<int128_t>(v1);
+        return r == l;
```
Comment (stdpain) [inline be/src/runtime/raw_value.h:159]:
```suggestion
        return r < l;
```

File: be/src/runtime/raw_value.h [hunk 4]
```diff
@@ -198,11 +204,17 @@ inline bool RawValue::eq(const void* v1, const void* v2, const TypeDescriptor& t
...
+        int128_t l = unaligned_load<int128_t>(v1);
+        return r < l;
+    }
 
-    case TYPE_LARGEINT:
-        return reinterpret_cast<const PackedInt128*>(v1)->value == reinterpret_cast<const PackedInt128*>(v2)->value;
+    case TYPE_LARGEINT: {
+        int128_t r = unaligned_load<int128_t>(v2);
+        int128_t l = unaligned_load<int128_t>(v1);
+        return r < l;
```
Comment (stdpain) [inline be/src/runtime/raw_value.h:216]:
```suggestion
        return r == l;
```

**Review by stdpain — DISMISSED** [review]
LGTM

File: be/src/runtime/raw_value.h [hunk 5]
```diff
@@ -147,11 +147,17 @@ inline bool RawValue::lt(const void* v1, const void* v2, const TypeDescriptor& t
     case TYPE_DECIMAL:
         return *reinterpret_cast<const DecimalValue*>(v1) < *reinterpret_cast<const DecimalValue*>(v2);
 
-    case TYPE_DECIMALV2:
-        return reinterpret_cast<const PackedInt128*>(v1)->value < reinterpret_cast<const PackedInt128*>(v2)->value;
+    case TYPE_DECIMALV2: {
+        int128_t r = unaligned_load<int128_t>(v2);
+        int128_t l = unaligned_load<int128_t>(v1);
+        return r < l;
```
Comment (imay) [inline be/src/runtime/raw_value.h:153]: should be 
```suggestion
        return l < r;
```

File: be/src/runtime/raw_value.h [hunk 6]
```diff
@@ -147,11 +147,17 @@ inline bool RawValue::lt(const void* v1, const void* v2, const TypeDescriptor& t
...
+    }
 
-    case TYPE_LARGEINT:
-        return reinterpret_cast<const PackedInt128*>(v1)->value < reinterpret_cast<const PackedInt128*>(v2)->value;
+    case TYPE_LARGEINT: {
+        int128_t r = unaligned_load<int128_t>(v2);
+        int128_t l = unaligned_load<int128_t>(v1);
+        return r < l;
```
Comment (imay) [inline be/src/runtime/raw_value.h:159]:
```suggestion
        return l < r;
```

**Review by imay — COMMENTED** [review]
@Ccuurryy Almost there, little change needed.

**Review by stdpain — APPROVED** [review]

**Comment by imay** [comment]
run starrocks_be_unittest

**Pull request merged**